backend/lcfs/web/api/compliance_report/repo.py

```diff
@@ -965,3 +956,58 @@ async def get_previous_summary(
             .limit(1)
         )
         return result.scalars().first().summary
+
+    async def get_latest_visible_reports_query(
+        self,
+        excluded_statuses: List[ComplianceReportStatusEnum],
+        user_organization_id=None,
+    ):
+        # Build a correlated subquery to determine the maximum version for each group,
+        # filtering only reports with a status not in the excluded list.
+
+        status_ids = []
+        for status in excluded_statuses:
+            status = await self.get_compliance_report_status_by_desc(status.value)
+            status_ids.append(status.compliance_report_status_id)
+
+        max_version_subq = (
+            select(func.max(ComplianceReport.version))
+            .where(
+                ComplianceReportListView.compliance_report_group_uuid
+                == ComplianceReport.compliance_report_group_uuid,
+                ComplianceReport.current_status_id.notin_(status_ids),
+            )
+            .scalar_subquery()
+        )
+
+        # Hide government reports unless they are assessed
+        if user_organization_id is not None:
+            assessed = await self.get_compliance_report_status_by_desc(
+                ComplianceReportStatusEnum.Assessed.value
+            )
+            max_version_subq = max_version_subq.where(
+                or_(
+                    ComplianceReport.current_status_id
+                    == assessed.compliance_report_status_id,
+                    or_(
+                        ComplianceReport.supplemental_initiator
+                        != SupplementalInitiatorType.GOVERNMENT_REASSESSMENT,
+                        ComplianceReport.supplemental_initiator.is_(None),
```

Comment:
I don't think we would ever have a null supplemental_initiator? or has that logic changed in this work?

Reply:
Supplemental Initiator is only set on supplementals currently. It is null for original reports.